src/pymovements/plotting/heatmap.py

```diff
@@ -0,0 +1,168 @@
+) -> plt.Figure:
+    """Plot a heatmap of gaze data.
+
+    The heatmap displays the distribution of gaze positions across the experiment screen,
+    for a given GazeDataFrame object.
+    The color values indicate the time spent at each position in seconds.
+
+    Parameters
+    ----------
+    gaze : GazeDataFrame
+        A GazeDataFrame object.
+    position_columns : tuple[str, str], optional
+        The column names of the x and y position data
+    gridsize : tuple[int, int], optional
+        The number of bins in the x and y dimensions.
+    cmap : colors.Colormap | str, optional
+        The colormap to use
+    interpolation : str, optional
+        The interpolation method to use for plotting the heatmap.
+        See matplotlib.pyplot.imshow for more information on available methods
+        for interpolation. By default, 'gaussian' is used.
+    origin : str, optional
+        origin of y-axis, valid values are 'lower' or 'upper'
+    figsize : tuple[float, float], optional
+        Figure size
+    cbar_label : str | None, optional
+        Label for the colorbar
+    show_cbar : bool, optional
+        Whether to show the colorbar.
+    title : str | None, optional
+        Figure title
+    xlabel : str | None, optional
+        x-axis label
+    ylabel : str | None, optional
+        y-axis label
+    show : bool, optional
+        Whether to show the plot
+    savepath : str | None, optional
+        If provided, the figure will be saved to this path
+
+    Raises
+    ------
+    ValueError
+        If the position columns are not in pixels or degrees
+    ValueError
+        If the experiment property of the GazeDataFrame is None
+    Returns
+    -------
+    plt.Figure
+        The heatmap figure.
+    """
+
+    # Extract x and y positions from the gaze dataframe
+    x = gaze.frame[position_columns[0]]
+    y = gaze.frame[position_columns[1]]
+
+    # Check if experiment properties are available
+    if not gaze.experiment:
+        raise ValueError(
+            'Experiment property of GazeDataFrame is None. '
+            'GazeDataFrame must be associated with an experiment.',
+        )
+
+    # Get experiment screen properties
+    screen = gaze.experiment.screen
+
+    # check if gaze data is in pixels or degrees
+    if 'pix' in position_columns[0] and 'pix' in position_columns[1]:
+        xmin, xmax = 0, screen.width_px
+        ymin, ymax = 0, screen.height_px
+    elif 'pos' in position_columns[0] and 'pos' in position_columns[1]:
+        xmin, xmax = screen.x_min_dva, screen.x_max_dva
+        ymin, ymax = screen.y_min_dva, screen.y_max_dva
+    else:
+        raise ValueError('Position columns must be in pixels or degrees')
+
+    # Define the grid and bin the gaze data
+    x_bins = np.linspace(xmin, xmax, num=gridsize[0])
+    y_bins = np.linspace(ymin, ymax, num=gridsize[1])
+
+    # Bin the gaze data
+    heatmap, x_edges, y_edges = np.histogram2d(x, y, bins=[x_bins, y_bins])
```

Comment:
naming the variable `heatmap` is problematic as it is also the function name.
our linters don't accept that.

name the variable `heatmap_values` or similar